feat: ✨ input validation using schema

mloiseleur opened the pull request:
### What does this PR do?

<!-- A brief description of the change being made with this pull request. -->


### Motivation

<!-- What inspired you to submit this pull request? -->


### More

- [ ] Yes, I updated the tests accordingly
- [ ] Yes, I ran `make test` and all the tests passed

<!--

HOW TO WRITE A GOOD PULL REQUEST? https://doc.traefik.io/traefik/contributing/submitting-pull-requests/

-->

mloiseleur commented:
@darkweaver87 @jnoordsij This PR can be drastically improved if we set default empty (and so _typed_) values.

It would mean the comment annotation with type would not be needed anymore.

darkweaver87 reviewed (approved):
LVGTM

mloiseleur commented:
@jnoordsij If you can take a look, it would be great :pray:

jnoordsij reviewed (approved):
Oooh I like this a lot, could be very useful to have for validation in IDEs to spot errors more easily. Great work!

Maybe we want to document this for contributors as well? Or even try to have some workflow setup to automatically update the generated schema file? Otherwise probably updating it somewhere in the release process similar to the changelog works fine as well, but still that should be noted somewhere.

mloiseleur commented:
Ah yes, I totally missed the update / workflow / contributor part. Thanks !

mloiseleur commented:
@jnoordsij @darkweaver87 I updated Guidelines and added a github action which fails when schema is not updated.

Anything to add / remove / improve before merge ?
